### PR TITLE
add github action for docker image push to ghcr registry

### PR DESCRIPTION
**Pull request by matprov**
- Adds GitHub Action for Docker image push to ghcr registry.
- Move the `npm build` to an entrypoint in order to allow us to customize base path (see `ROOT_PATH`) and catalog URL (see `CATALOG_URL`), which are compiled for the frontend. Note that this change makes the startup time longer, since `npm build` needs to be executed on container boot.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            echo "::set-output name=TAG_VERSION::latest"
+          else
+            echo "::set-output name=TAG_VERSION::${GITHUB_REF##*/}"
+          fi
+      - name: Extract branch name
+        id: extract_branch
+        shell: bash
+        run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
+      - name: Log in to the container registry
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
```
Comment (waystilos) [inline .github/workflows/release.yml:35]: What's the reasoning for not using the latest (`docker/login-action@v2`)?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      # - name: Build and push image using tag
```
Comment (waystilos) [inline .github/workflows/release.yml:40]: Does this need remain since it's commented out?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+name: Release Docker image
```
Comment (waystilos) [inline .github/workflows/release.yml:1]: NIT: Capitalize Image to stay consistent.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      # - name: Build and push image using tag
+      #   uses: docker/build-push-action@v3
+      #   with:
+      #     context: .
+      #     push: true
+      #     tags: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ steps.version.outputs.TAG_VERSION }}
+      - name: Build and push image using branch name
+        uses: docker/build-push-action@v3
```
Comment (waystilos) [inline .github/workflows/release.yml:47]: You should be able to update this to v4: `docker/build-push-action@v4`

**Comment by m-mohr** [comment]
@matprov The dev branch got deleted in favor of main. GitHub closed the PR automatically due to this. Please re-open the PR for the main branch.

**Pull request closed without merging**